linode/instance/resource.go

```diff
@@ -583,6 +587,26 @@ func updateResource(ctx context.Context, d *schema.ResourceData, meta interface{
 		simpleUpdate = true
 	}
 
+	if d.HasChange("placement_group.0.id") {
+		oldPGID, newPGID := d.GetChange("placement_group.0.id")
+
+		tflog.Debug(ctx, "Attempting Placement Group reassignment", map[string]any{
+			"old_pg_id": oldPGID,
+			"new_pg_id": newPGID,
+		})
+
+		if err := reassignPlacementGroup(
+			ctx,
+			client,
+			instance.ID,
+			oldPGID.(int),
+			newPGID.(int),
+			helper.SDKv2UnwrapOptionalConfigAttr[bool](ctx, d, "placement_group.0.compliant_only"),
```

Comment:
This is more or less the same thing as how we check that the user has defined the `booted` attribute, but it needs a bit more complexity since `compliant_only` is deeply nested.